### PR TITLE
Add '--cert-identity' flag to support subject alternate names for ver…

### PR DESCRIPTION
**Pull request by kpk47**
…ification.

#### Summary

This PR adds the '--cert-identity' flag for supporting SANs during verification. This change is working towards #2056, but does not fix it completely.

Fixes sigstore/cosign#1964

#### Release Note

Added support for certificate Subject Alternate Names during verification flow.

#### Documentation

N/A, though this flag will need documenting if/when it completely replaces --cert-email.

### CODE REVIEW

**Comment by maxking** [comment]
You probably need to run `make docgen` to regenerate the markdown documentation for the CLI. It seems to be causing CI to fail.

File: cmd/cosign/cli/options/certificate.go [hunk 1]
```diff
@@ -43,6 +44,9 @@ func (o *CertVerifyOptions) AddFlags(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&o.CertEmail, "certificate-email", "",
 		"the email expected in a valid Fulcio certificate")
 
+	cmd.Flags().StringVar(&o.CertIdentity, "certificate-identity", "",
+		"the identity expected in a valid Fulcio certificate")
```
Comment (znewman01) [inline cmd/cosign/cli/options/certificate.go:48]: I'd like to see a little bit more detail in this help text. What's an "identity" vs. an "email"?

**Review by znewman01 — APPROVED** [review]

File: cmd/cosign/cli/options/certificate.go (hunk 1 above)
Comment (kpk47) [reply cmd/cosign/cli/options/certificate.go]: I added a list of acceptable types of values. WDYT?

**Review by znewman01 — APPROVED** [review]
LGTM, thank you so much!

(I don't actually have CODEOWNERs here but someone with CODEOWNERs will swing by soon)

File: cmd/cosign/cli/options/certificate.go (hunk 1 above)
Comment (znewman01) [reply cmd/cosign/cli/options/certificate.go]: Looks great!

**Review by haydentherapper — APPROVED** [review]
very nice!

**Comment by znewman01** [comment]
@kpk47 can you add the DCO to your last commit?

**Review by znewman01 — APPROVED** [review]

**Review by haydentherapper — DISMISSED** [review]

**Comment by haydentherapper** [comment]
@kpk47 You'll need to sign the DCO again.

**Review by znewman01 — APPROVED** [review]

**Pull request merged**